courses/tests.py

```diff
@@ -46,5 +53,104 @@ def test_category_get_view(self):
             ]
         })
         self.assertEqual(response.status_code, 200)
+class CourseDetailViewTest(TestCase):
+    def setUp(self):
+        User.objects.create(
+            id          = 1,
+            email       = "abc@naver.com",
+            korean_name = "철수",
+            password    = "12345",
+            kakao_id    = 1
+        )
+        Category.objects.create(
+            id   = 1,
+            name = "강의"
+        )
+        SubCategory.objects.create(
+            id          = 1,
+            name        = "미술",
+            category_id = 1
+        )
+        Target.objects.create(
+            id   = 1,
+            name = "초급자용"
+        )
+        Course.objects.create(
+            id              = 1,
+            title           = "드로잉",
+            price           = 240000,
+            description     = "abcd",
+            thumbnail       = "abcd.jpg",
+            month           = 6,
+            sub_category_id = 1,
+            target_id       = 1,
+            user_id         = 1
+        )
+        Like.objects.create(
+            id        = 1,
+            course_id = 1,
+            user_id   = 1
+        )
+        Review.objects.create(
+            id        = 1,
+            text      = "hihi",
+            course_id = 1,
+            user_id   = 1
+        )
+    
+    def tearDown(self):
+        Course.objects.all().delete()
 
+    def test_CourseDetailView_get_success(self):
+        client          = Client()
+        response        = client.get('/courses/1')
+        course          = Course.objects.get(id=1)
+        course_like     = Like.objects.filter(course_id=1)
+        course_question = Review.objects.filter(course_id=1)
+        answers         = []
+        for question_id in course_question:
+            answers.append(question_id.id)
+        question_answer = Comment.objects.filter(review_id__in=answers)
+        course_info     = {
+                "id"            : course.id,
+                "name"          : course.title,
+                "price"         : "{:.2f}".format(course.price),
+                "thumbnail_url" : course.thumbnail,
+                "subcategory"   : course.sub_category.name,
+                "counts_like"   : course_like.count(),
+                "target"        : course.target.name,
+                "month"         : course.month,
+                "question"      : list(course_question.values('id', 'text', 'user_id')),
+                "answer"        : list(question_answer.values('text', 'user_id', 'review_id'))
```

Comment:
list comprehension 을 사용해주시고, 현재와 같은 방식외에, 아래와 같은 방식으로 작성해주세요.
```suggestion
                "question"      : {
                               'id' : id,
                               'text' : text
                 }
```

Reply:
list comprehension 사용하여 코드 수정하였습니다.